go/obscuronode/host/host.go

```diff
@@ -51,6 +51,7 @@ type Node struct {
 	ID common.Address
 
 	P2p          P2P
+	P2pAddress   string
```

Comment:
what's up with this field?
Seems to be used for testing only.  
Anyway, it should be in the test itself.

Reply:
I thought this was consistent with other fields on `Node`. For example, the `Enclave` and `P2p` fields are exposed for testing purposes only. I thought this was cleaner than passing around the node addresses separately from the node objects.

Reply:
Umm. But this field is only for testing. It has no meaning in the production code alone. I'd say we find another way to implement the functionality